kornia/geometry/epipolar/fundamental.py

```diff
@@ -128,6 +329,32 @@ def find_fundamental(
     return normalize_transformation(F_est)
 
 
+def find_fundamental(
+    points1: Tensor, points2: Tensor, weights: Optional[Tensor] = None, method: Literal['8POINT', '7POINT'] = '8POINT'
+) -> Tensor:
+    r"""
+    Args:
+        points1: A set of points in the first image with a tensor shape :math:`(B, N, 2), N>=8`.
+        points2: A set of points in the second image with a tensor shape :math:`(B, N, 2), N>=8`.
+        weights: Tensor containing the weights per point correspondence with a shape of :math:`(B, N)`.
+        method: The method to use for computing the fundamental matrix. Supported methods are "7POINT" and "8POINT".
+
+    Returns:
+        the computed fundamental matrix with shape :math:`(B, 3*m, 3)`, where `m` number of fundamental matrix.
+
+    Raises:
+        ValueError: If an invalid method is provided.
+
+    """
+    if method.upper() == "7POINT":
+        result = run_7point(points1, points2)
```

Comment:
Will this work for batches as stated in the docs ?

Reply:
need to make some fixes in order to enable batch processing. I'm currently looking into it

Reply:
@anandhupvr how about adding the test case first and check it?

Reply:
[20c02fd5469c29645e08f86467e52c0a6d8e4cfc](https://github.com/kornia/kornia/pull/2390/commits/20c02fd5469c29645e08f86467e52c0a6d8e4cfc) added a shape test